packages/portal/backend/src/server/common/ClasslistAgent.ts

```diff
@@ -0,0 +1,150 @@
+                rejectUnauthorized: false
+            };
+            return JSON.parse(await rp(options));
+        } catch (err) {
+            Log.error("IntegrationController::fetchClasslist - ERROR: " + err);
+            throw new Error("Could not fetch Classlist " + err.message);
+        }
+    }
+
+    private getClasslistUri() {
+        const config = Config.getInstance();
+        const auth = config.getProp(ConfigKey.classlist_username) + ':' + config.getProp(ConfigKey.classlist_password);
+        const uri = config.getProp(ConfigKey.classlist_uri);
+
+        if (uri.indexOf('https://') === 0) {
+            return 'https://' + auth + '@' + uri.slice(8);
+        } else {
+            throw new Error('https:// protocol is required for API integration');
+        }
+    }
+
+    public async processClasslist(personId: string = null, path: string = null,  data: any): Promise<Person[]> {
+        Log.trace("PersonController::processClasslist(...) - start");
+
+        if (path !== null) {
+            data = await new CSVParser().parsePath(path);
+        }
+
+        this.duplicateDataCheck(data, ['ACCT', 'CWL']);
+        this.missingDataCheck(data, ['ACCT', 'CWL']);
+        const peoplePromises: Array<Promise<Person>> = [];
+
+        for (const row of data) {
+            // Log.trace(JSON.stringify(row));
+            if (typeof row.ACCT !== 'undefined' && typeof row.CWL !== 'undefined' &&
+                typeof row.SNUM !== 'undefined' && typeof row.FIRST !== 'undefined' &&
+                typeof row.LAST !== 'undefined' && typeof row.LAB !== 'undefined') {
+                const p: Person = {
+                    id:            row.ACCT.toLowerCase(), // id is CSID since this cannot be changed
+                    csId:          row.ACCT.toLowerCase(),
+                    // github.ugrad.cs wanted row.ACCT; github.students.cs and github.ubc want row.CWL
+                    githubId:      row.CWL.toLowerCase(),
+                    studentNumber: row.SNUM,
+                    fName:         row.FIRST,
+                    lName:         row.LAST,
+
+                    kind:   PersonKind.STUDENT,
+                    URL:    null,
+                    labId:  row.LAB,
+                    custom: {}
+                };
+                peoplePromises.push(this.pc.createPerson(p));
+            } else {
+                Log.error('PersonController::processClasslist(..) - column missing from: ' + JSON.stringify(row));
+                peoplePromises.push(Promise.reject('Required column missing (required: ACCT, CWL, SNUM, FIRST, LAST, LAB).'));
+        }
+    }
+        const people = await Promise.all(peoplePromises);
+
+        // audit
+        await this.db.writeAudit(AuditLabel.CLASSLIST_UPLOAD, personId,
+            {}, {}, {numPoeple: people.length});
+
+        return people;
+    }
+
+    private duplicateDataCheck(data: any[], columnNames: string[]) {
+        Log.trace('CSVParser::duplicateDataCheck -- start');
```

Comment:
As above the log messages should refer to the right file name.

Reply:
Agreed